server/app/playbook_run_service.go

```diff
@@ -1656,10 +1656,10 @@ func (s *PlaybookRunServiceImpl) MoveChecklist(playbookRunID, userID string, sou
 	checklistMoved := playbookRunToModify.Checklists[sourceChecklistIdx]
 
 	// Delete checklist to move
-	playbookRunToModify.Checklists = append(playbookRunToModify.Checklists[:sourceChecklistIdx], playbookRunToModify.Checklists[sourceChecklistIdx+1:]...)
+	copy(playbookRunToModify.Checklists[sourceChecklistIdx:], playbookRunToModify.Checklists[sourceChecklistIdx+1:])
+	playbookRunToModify.Checklists[len(playbookRunToModify.Checklists)-1] = Checklist{}
```

Comment:
You'll also want to shorten the checklist, right?
```suggestion
	playbookRunToModify.Checklists[len(playbookRunToModify.Checklists)-1] = Checklist{}
	playbookRunToModify.Checklists = playbookRunToModify.Checklists[:len(playbookRunToModify.Checklists)-1]
```

Reply:
@cpoile: Immediately after the deletion we insert the new element, so instead of shortening the checklist and then appending a new item:

```go
// Delete
copy(c[i:], c[i+1:])
c[len(c)-1] = Checklist{}
c = c[:len(c)-1]

// Insert
c = append(c, Checklist{})
copy(c[j+1:], c[j:])
c[j] = checklistMoved
```

I simply keep the same length and copy the elements to their position:

```go
// Delete
copy(c[i:], c[i+1:])
c[len(c)-1] = Checklist{}

// Insert
copy(c[j+1:], c[j:])
c[j] = checklistMoved
```

Does that make sense?

Reply:
ah, I was afraid I overlooked something, and that was it. Thanks!